lib/ssh_scan/policy.rb

```diff
@@ -25,5 +26,20 @@ def self.from_string(string)
       opts = YAML.load(string)
       self.new(opts)
     end
+
+    # Choose a policy file based on rules
+    def self.first_match(result)
+      Dir.glob("policies/**/*.yml") do |policy_file|
+        next if policy_file == @fallback_file
+        file = YAML.load_file(policy_file)
+        next unless file["rules"]
+        puts "Matching: #{policy_file}"
+        if SSHScan::RuleEngine.match(file["rules"], result)
+          puts "Matched: #{policy_file}"
+          policy_file
```

Comment:
In ruby, you get an implicit return if it's the last operation in the scope.

def example
   foo #will return foo, but implicitly
end

def example
   return foo #will also return foo, but explicitly
end

but if you want an early return, you need to be explicit...

def example
  if foo
    return bar #if foo, then return bar
  end

  foo # if not foo, then default to implicit return of foo
end